File: src/app/components/post/dto/post.ts
export class PostDTO {

  postID;
  post;
  feeling;
  userID;
  userName;
  bookName;

  constructor(posts?:any){
    posts = posts || {};
    this.postID = posts.postID || null;
    this.bookName = posts.bookName || '';
    this.post = posts.post || '';
    this.feeling = posts.feeling || '';
    this.userID = posts.userID || null;
    this.userName = posts.userName || '';
  }
}

File: backend/main/java/com/example/student/model/Post.java
package com.example.student.model;

import javax.persistence.*;

@Entity
@Table(name = "t_post")
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "POST_ID",nullable = false,updatable = false)
    private Integer postID;

    @Column(name = "POST")
    private String post;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "USER_ID")
    private UserPost user;

    @Column(name = "FEELING")
    private String feeling;

    @Column(name = "BOOK_Name")
    private String bookName;

    public Integer getPostID() {
        return postID;
    }

    public void setPostID(Integer postID) {
        this.postID = postID;
    }

    public String getPost() {
        return post;
    }

    public void setPost(String post) {
        this.post = post;
    }

    public UserPost getUser() {
        return user;
    }

    public void setUser(UserPost user) {
        this.user = user;
    }

    public String getFeeling() {
        return feeling;
    }

    public void setFeeling(String feeling) {
        this.feeling = feeling;
    }

    public String getBookName() {
        return bookName;
    }

    public void setBookName(String bookName) {
        this.bookName = bookName;
    }
}
